safe/gui/analysis_utilities.py

```diff
@@ -252,26 +253,54 @@ def add_layers_to_canvas_with_custom_orders(order, impact_function):
     :param impact_function: The multi exposure impact function used.
     :type impact_function: MultiExposureImpactFunction
     """
-    # Need to remove all groups and layers
-    QgsMapLayerRegistry.instance().removeAllMapLayers()
     root = QgsProject.instance().layerTreeRoot()
-    root.removeAllChildren()
+    root.setVisible(False)  # Make all layers hidden.
 
-    # Insert layers in the good order.
-    for layer in order:
-        if layer[0] == FROM_CANVAS['key']:
+    group_analysis = root.insertGroup(0, impact_function.name)
```

Comment:
do we have a definition for multi exposure string format? because I use raw string comparison in my PR for identifying a group whether it's multi exposure group or not

Reply:
ouch, doing string comparison is very dangerous. I'm adding a flag on the group for you. Wait 5 minutes